pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -218,25 +219,19 @@ def gather_data(seq: pulser.sequence.Sequence, gather_output: bool) -> dict:
                     eom_block_n + 1 < nb_eom_intervals
                     and slot.tf == eom_intervals[eom_block_n + 1].ti
                     and extended_samples.det[slot.tf - 1]
-                    == sch.eom_blocks[eom_block_n + 1].detuning_off
+                    == ch_samples.eom_blocks[eom_block_n + 1].detuning_off
                 ):
                     # Buffer if next is eom and final det matches det_off
                     eom_start_buffers[eom_block_n + 1] = EOMSegment(
                         slot.ti, slot.tf
                     )
 
-            if slot.type == "target":
-                target[(slot.ti, slot.tf - 1)] = slot.targets
+        for time_slot in ch_samples.target_time_slots:
+            if time_slot.ti == -1:
+                target["initial"] = time_slot.targets
                 continue
-            if slot.type == "delay":
-                continue
-            pulse = cast(Pulse, slot.type)
-            for wf_type in ["amplitude", "detuning"]:
-                wf = getattr(pulse, wf_type)
-                if isinstance(wf, InterpolatedWaveform):
-                    pts = wf.data_points
-                    pts[:, 0] += slot.ti
-                    interp_pts[wf_type] += pts.tolist()
+            if time_slot.type == "target":
```

Comment:
Aren't they all supposed to be of type "target" ? ;)
It's almost an assert

Reply:
yep, totally useless!